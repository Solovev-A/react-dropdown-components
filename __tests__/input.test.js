import React from 'react';
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom';
import 'jest-styled-components';

import Input from '../lib/components/input';

describe('Input component', () => {
    it('renders correctly', () => {
        render(<Input />);
        expect(screen.getByRole('textbox')).toMatchSnapshot();
    });

    describe('behavior', () => {
        it('use callback on search change', () => {
            const onChange = jest.fn();
            render(<Input onSearchChange={onChange} />);

            userEvent.type(screen.getByRole('textbox'), 'test');

            expect(onChange).toHaveBeenCalledTimes(4);
            expect(onChange).toBeCalledWith('test');
        });

        it('use value prop', () => {
            render(<Input value="test" />);
            expect(screen.getByRole('textbox')).toHaveValue('test');
        });
    });
});