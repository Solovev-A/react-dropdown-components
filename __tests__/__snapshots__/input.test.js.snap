// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-top: calc(.375rem - 2px);
  padding: 0;
  box-sizing: border-box;
  border: none;
  font-family: inherit;
  font-size: inherit;
  background-color: inherit;
  line-height: inherit;
  color: #495057;
}

.c0:focus {
  outline: 0;
}

<input
  class="c0"
  value=""
/>
`;
